admin/admin.go

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2024 The VeChainThor developers
+
+// Distributed under the GNU Lesser General Public License v3.0 software license, see the accompanying
+// file LICENSE or <https://www.gnu.org/licenses/lgpl-3.0.html>
+
+package admin
+
+import (
+	"encoding/json"
+	"fmt"
+	"log/slog"
+	"net/http"
+
+	"github.com/gorilla/handlers"
+	"github.com/gorilla/mux"
+	"github.com/vechain/thor/v2/log"
+)
+
+type logLevelRequest struct {
+	Level string `json:"level"`
+}
+
+type logLevelResponse struct {
+	CurrentLevel string `json:"currentLevel"`
+}
+
+func getLogLevelHandler(logLevel *slog.LevelVar) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		response := logLevelResponse{
+			CurrentLevel: logLevel.Level().String(),
+		}
+		w.Header().Set("Content-Type", "application/json")
+		if err := json.NewEncoder(w).Encode(response); err != nil {
+			http.Error(w, "Failed to encode response", http.StatusInternalServerError)
```

Comment:
I think it would be nice to have an always json-format responses.
For payloads, responses and error responses everything as a json, seems to me, that would make easier than a json sometimes and a plain text on other times ?